### PR TITLE
Updates for api 1.33 from cleanflight

### PR DESCRIPTION
**Pull request by mikeller**

### CODE REVIEW

**Comment by hydra** [comment]
I'll have a look when I get a chance.  Thanks for prepping it.

**Comment by mikeller** [comment]
Fixes #486.

**Comment by TheAngularity** [comment]
i'm added two new fixes -> https://github.com/cleanflight/cleanflight-configurator/pull/474 and https://github.com/cleanflight/cleanflight-configurator/pull/475

**Comment by mikeller** [comment]
@TheAngularity Thanks, cherry-picked them.

File: js/backup_restore.js [hunk 1]
```diff
@@ -527,11 +529,11 @@ function configuration_restore(callback) {
...
             // api 1.14 exposes 3D configuration
             
-            if (configuration._3D == undefined) {
-                configuration._3D = {
+            if (configuration.MOTOR_3D_CONFIG == undefined) {
+                configuration.MOTOR_3D_CONFIG = {
                     deadband3d_low:         1406,
                     deadband3d_high:        1514,
-                    neutral3d:              1460,
+                    neutral:              1460,
```
Comment (martinbudden) [inline js/backup_restore.js]: Whitespace

File: js/backup_restore.js [hunk 2]
```diff
@@ -566,9 +568,9 @@ function configuration_restore(callback) {
                 configuration.RX_CONFIG = {
                     serialrx_provider:      0,
                     spektrum_sat_bind:      0,
-                    midrc:                  1500,
-                    mincheck:               1100,
-                    maxcheck:               1900,
+                    stick_center:                  1500,
```
Comment (martinbudden) [inline js/backup_restore.js]: whitespace

**Comment by martinbudden** [comment]
Looks good to me.

**Comment by TheAngularity** [comment]
tested and issues found :( 
first issue is that the expert mode not working correct ... i'm disabled the expert mode but after connect all tabs in the tab-list are shown ... after klick expert mode on and off the hidden tabs in tabs-list go away ... but all tabs are only displayed correctly when i view the configurator tab

second issue comes when i'm exit from cli ... this ends in an empty tab
exitfromcli

after connect you can see this now
newconnect

only a restart of the configurator fixes it

**Comment by hydra** [comment]
I deleted expert mode in CF last week. I never liked it and since it was broken I found it was a better solution than trying to fix it.  If you want to keep it then you'll need to make sure you fetch the features on connect but only if the board type is not an OSD slave.

**Comment by hydra** [comment]
Also, there were various fixes to the serial port handling code in CF configurator master, pull them  too.

**Comment by mikeller** [comment]
@hydra: What was broken with expert mode?

**Comment by mikeller** [comment]
Ok, cherry-picked appropriate commits from cleanflight-configurator/master, reinstated expert mode.

**Comment by TheAngularity** [comment]
ok, tested now latest merged commits ... the issue with exit from cli is now fixed ... issue with expertmode is still available

**Comment by mikeller** [comment]
@TheAngularity: I see now what you mean. It wasn't expert mode, it was the tabs for non-activated features not being hidden. Fixed.

**Comment by TheAngularity** [comment]
ok, will test ... are you read my latest edits

**Comment by hydra** [comment]
@mikeller yeah, it's not difficult to fix, i just didn't like the implementation of expert mode at all.  My reasons were on the commit message that removed it.

https://github.com/cleanflight/cleanflight-configurator/commit/97089c182872249cb4471d3e7812af8d92ff797d

I'll re-cap here:
 CF - Delete expert mode.

It was:

a) badly implemented since lots of expert settings were still available.
b) hid important things pilots should know about (e.g. failsafe!)
c) it didn't play nice with api 1.33.0+ due to removal of msp features.
d) slider was not correctly styled all the time.
e) it messed up the google analytics feature, some code was left behind.

**Comment by TheAngularity** [comment]
@mikeller 
latest commit looks good ... expert mode working correct now

now only the cf color scheme and glitches on dataflash info ... imho redo 34cd69e fixed it

**Comment by mikeller** [comment]
@hydra: I never merged that one or read the comments on it, since it was marked 'CF' (and I am still backlogged).

Expert mode was implemented after a lengthy discussion about the problem of newbies changing settings that should be changed only for special cases, and ending up with not flyable / unsafe setups, vs. experts wanting to be able to configure all of the aspects of their craft from the GUI. 

I think the points that you make are mostly correct, but can also easily be addressed:
a) it was never fully implemented, since the granularity only ever went down to tab level, and not setting level;
b) failsafe was exactly what started the discussion that ended in the decision to implement expert mode: The default settings for failsafe are chosen so that they are what is the safest option for non-experienced pilots to use. Having the failsafe tab show led newbies to think they had to change them;
c) `MSP.send_message(MSPCodes.MSP_FEATURE_CONFIG, false, false);` in `serial_backend.js` fixes this;
d) would have been easy to fix (I think);
e) no idea. that code was removed before I started working on the Betaflight configurator, and I never came across any code that was left behind.

**Comment by mikeller** [comment]
@TheAngularity: 34cd69e is in my PR :-|

**Comment by TheAngularity** [comment]
@mikeller 
yep ... imho you should undo these changes

**Comment by mikeller** [comment]
@TheAngularity: Try now.

**Comment by TheAngularity** [comment]
@mikeller 
looks good again ... i'll check out if i find something else

**Comment by TheAngularity** [comment]
@mikeller @hydra 
unfortunately there is still something to repair ... at this time im testing bf 3.2.0 and all is good ... but the configuration is not downward compatible with bf 3.1.7 and feature vbat / current meter ... config boxes for vbat / current meter are not displayed in configuration tab and the power tab is displayed this:
zwischenablagebild 2

imho the boxes voltage and amperage should be not displayed on this warning

**Comment by mikeller** [comment]
Yes, that is a bit of a downer. For Betaflight we need something that works pre  and post new battery code.

@hydra: Also, the error message is wrong, it states 1.22 as the version break for the new battery code.

**Comment by TheAngularity** [comment]
@mikeller 
found now a fix for the appeared voltage and amperage box if the warning displayed--> made a new pr i your repo 637cbdb

**Comment by mikeller** [comment]
@TheAngularity: Unfortunately just not displaying the boxes won't help users of 3.1.7, since that does not allow them to configure their battery settings. :-(

**Comment by TheAngularity** [comment]
@mikeller 
Should also be only a fix for the appeared boxes when the warning displayed ;))

**Comment by TheAngularity** [comment]
@mikeller @hydra 
so, i again ... other errors i have not found now ... only the compatibility issue with vbat / current meter should be eliminated ... then everything would be all right again ;))

**Comment by hydra** [comment]
IMHO you can ignore the compatibility with older versions.  Battery/Voltage/Current settings are somehing you setup once, and leave alone, when you setup a new quad.  people with existing firmware won't need to do this very often.  If they do, they can still use the CLI.  This is what I decided at the time for the CF configurator.   For CF configurator it makes no sense at all to add backwards compatibility because there are no old versions out there because CF configurator dropped support for anything less than 2.x so there was no issue.

I think we've got bigger fish to fry, such as adding mixer configuration tab and working on the lack of chrome store in a few months and I'd rather we worked on that than providing support for the few users of old firmware who might need to change the setting once or twice.

**Comment by TheAngularity** [comment]
@hydra i agree with you ... have only thought about the bf 3.2.0 has not yet released ... i myself have no problem with this i currently use the bf 3.2.0 and will not go back to 3.1.7 ... if @mikeller is the same opinion then this can remain so and it could be merged

**Comment by mikeller** [comment]
@hydra: I am not the biggest fan of dropping compatibility with older versions if it can be avoided, since experience shows that there is always a proportion of people who prefer to stay with 'tried and true'. But in this case, the extra effort required for this makes it look like it's not gonna be worth it.
What's your opinion on dropping support for GUI battery configuration for <= 3.1.7, @borisbstyle, @martinbudden, @blckmn, @nathantsoi?

**Comment by martinbudden** [comment]
I'm also not a fan of dropping compatibility with older versions. But in this case I agree it does not seem worth it.

In my view the chrome store issue should be the absolute top priority for the configurator.

**Comment by nathantsoi** [comment]
Sry, I missed the @mention. I agree with @martinbudden. I do think dropping 3.1.7 support, if I'm reading that right, is going to be really painful.

**Comment by nathantsoi** [comment]
Also seems like there were only 2 msp changes in firmware, why is this patch set so huge? Also... I already did the non chrome store fix w/o dropping compatibility. How is that related @hydra?

**Comment by mikeller** [comment]
@nathantsoi: Agreed that dropping support for 3.1.7 battery configuration will be painful. This should only be released to the Chrome Web store once 3.2.0 is being released as well. In the meantime, users using latest builds will have to use the configurator from the repo as well - at least everything is working this way now.

Fixing the MSP changes for betaflight-configurator with separate commits did not look like an attractive option to me, since this would create additional effort to resolve merge conflicts for future cherry-picks from cleanflight-configurator.

**Pull request merged**